fix(linter): `unicorn/prefer-array-flat-map` ignore `React.Children`

Sysix opened the pull request:
The original rule is ignoring this case:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d12fc01c9d7d089f0445df6027b7bd318ab38955/rules/prefer-array-flat-map.js#L10

Added the same test cases:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d12fc01c9d7d089f0445df6027b7bd318ab38955/test/prefer-array-flat-map.js#L32-L34

closes #13531

Sysix commented:
* **#13534** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13534?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13534?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13534).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/rules/unicorn/prefer_array_flat_map.rs:
```diff
@@ -43,6 +43,9 @@ declare_oxc_lint!(
 );
 
+// skip React.Children because we are only look after `StaticMemberExpression.property` and not its object
+const IGNORE_OBJECTS: [&str; 1] = [/* "React.Children", */ "Children"];
```
That why the comment above is there

camc314 commented:
### Merge activity

* **Sep 4, 5:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
fix(linter): `unicorn/prefer-array-flat-map` ignore `React.Children` (#13534)

The original rule is ignoring this case:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d12fc01c9d7d089f0445df6027b7bd318ab38955/rules/prefer-array-flat-map.js#L10

Added the same test cases:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d12fc01c9d7d089f0445df6027b7bd318ab38955/test/prefer-array-flat-map.js#L32-L34

closes #13531
